### PR TITLE
Add Convenience Constructors to shapes

### LINKED ISSUE #100: Convenience constructors for shapes
Given a `PointF` arrays `contour` and `hole`, this is how a `ComplexPolygon` could be constructed today:

```C#
ComplexPolygon polygon = new ComplexPolygon(
                new Path(new LinearLineSegment(contour)),
                new Path(new LinearLineSegment(hole)));
```

It should be as easy as:
```C#
ComplexPolygon polygon = new ComplexPolygon(contour, hole);
```

We need to review the shape types and add convenience constructors based on the principle of "good defaults".

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp.Drawing/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Fix #100

### CODE REVIEW

File: src/ImageSharp.Drawing/Shapes/Path.cs [hunk 1]
```diff
@@ -17,6 +17,15 @@ public class Path : IPath, ISimplePath, IPathInternals, IInternalPathOwner
         private readonly ILineSegment[] lineSegments;
         private InternalPath innerPath;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Path"/> class.
+        /// </summary>
+        /// <param name="points">The collection of points.</param>
```
Comment (tocsoft) [inline src/ImageSharp.Drawing/Shapes/Path.cs:23]: worth pointing out these are processed as a series of linear line segments

File: src/ImageSharp.Drawing/Shapes/Polygon.cs [hunk 2]
```diff
@@ -11,6 +11,15 @@ namespace SixLabors.ImageSharp.Drawing
     /// </summary>
     public class Polygon : Path
     {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Polygon"/> class.
```
Comment (tocsoft) [inline src/ImageSharp.Drawing/Shapes/Polygon.cs:15]: again woth pointing out thye are handled as a LinearLineSegment

File: src/ImageSharp.Drawing/Shapes/Path.cs (hunk 1 above)
Comment (JimBobSquarePants) [reply src/ImageSharp.Drawing/Shapes/Path.cs]: done

File: src/ImageSharp.Drawing/Shapes/Polygon.cs (hunk 2 above)
Comment (JimBobSquarePants) [reply src/ImageSharp.Drawing/Shapes/Polygon.cs:15]: and done

**Review by tocsoft — APPROVED** [review]

**Pull request merged**